inc/ticket.class.php

```diff
@@ -4865,18 +4867,60 @@ function showForm($ID, $options = []) {
          $content = Html::cleanPostForTextArea($content);
       }
 
-      $content = Html::setRichTextContent(
-         $content_id,
-         $content,
-         $rand,
-         !$canupdate
-      );
```

Comment:
Cannot call this method because the inner call to Html::clean() breaks reload of an invalid form submission

Reply:
All changes on the way rich text is handled always have been consequences....
I'm still against any changes on that.

Reply:
call to setRichTextContent is replaced by its content, with one difference : the call to Html::clean() below is done only if the ticket is displayed from an instace saved in DB. In other cases it is displayed from a user input (after a refresh of the form or after the form is displayed again after a mandatory field is not properly filled (i.e : category).

Reply:
Whatever changes made here, it should be reported to Problem and Change forms.

Reply:
We could apply following patch:
```diff
diff --git a/inc/html.class.php b/inc/html.class.php
index 180ffba60b..ae6be92125 100644
--- a/inc/html.class.php
+++ b/inc/html.class.php
@@ -95,6 +95,13 @@ class Html {
          ]
       );
 
+      // Special case : remove the 'denied:' for base64 img in case the base64 have characters
+      // combinaison introduce false positive
+      foreach (['png', 'gif', 'jpg', 'jpeg'] as $imgtype) {
+         $value = str_replace('src="denied:data:image/'.$imgtype.';base64,',
+               'src="data:image/'.$imgtype.';base64,', $value);
+      }
+
       $value = str_replace(["\r\n", "\r"], "\n", $value);
       $value = preg_replace("/(\n[ ]*){2,}/", "\n\n", $value, -1);
 
```

This is already done on `Toolbox::unclean_html_cross_side_scripting_deep()`. Should be mutualized.